Src/library/base/elks/kernel/routine.e

```diff
@@ -154,12 +180,14 @@ feature -- Status report
 							Result := arg = Void or else
 								field_conforms_to ({ISE_RUNTIME}.dynamic_type (arg), l_type)
 						end
+					else
+						Result := False
 					end
 					i := i + 1
 				end
 			end
-			if Result and then not is_target_closed and then args /= Void then
-				Result := args.item (1) /= Void
+			if Result and then not is_target_closed then
+				Result := args /= Void and then not args.is_empty and then args.item (1) /= Void
```

Comment:
There is no assertion which states that:
   is_target_closed implies closed_count >  0
   not is_target_closed implies open_count >  0
Without these assertions, the new implementation above is more accurate.